src/transformers/generation_utils.py

```diff
@@ -431,11 +431,16 @@ def generate(
                 .view(-1)
                 .to(input_ids.device)
             )
+
             # expand encoder_outputs
-            encoder_outputs = (encoder_outputs[0].index_select(0, expanded_batch_idxs), *encoder_outputs[1:])
+            encoder_outputs["last_hidden_state"] = encoder_outputs.last_hidden_state.index_select(
+                0, expanded_batch_idxs
+            )
+
+            # save encoder_outputs in `model_specific_kwargs`
+            model_specific_kwargs["encoder_outputs"] = encoder_outputs
```

Comment:
(nit/scope)  Think `model_specific_kwargs` should be called `forward_kwargs`/`model_kwargs`.